tests/baselines/reference/emptyArrayBindingPatternParameter04.js

```diff
@@ -0,0 +1,11 @@
+//// [emptyArrayBindingPatternParameter04.ts]
+
+
+function f([] = [1,2,3,4]) {
```

Comment:
sorry it should not be an error. it should be emitted regardless of the varible being used or not..

```ts
function f([] = [1,somefunctionWithSideEffects()]) {
}

// should emit
function f(_a) {
    var _b = _a === void 0 ? [1, someFunctionWithSideEffects()] : _a;
}
```

similar to how the var case you mentioned works today..

```ts
let [] = [1, someFunctionWithSideEfects()]; 

// emits
var _a = [1, someFunctionWithSideEfects()];
```